taipy/core/data/data_node.py

```diff
@@ -605,6 +608,31 @@ def get_last_edit(self) -> Optional[Edit]:
         """
         return self._edits[-1] if self._edits else None
 
+    def _get_rank(self, scenario_config_id: str) -> int:
+        """Get the data node rank for given scenario config.
+
+        The rank corresponds to the order of appearance of the data nodes in a scenario config DAG.
+
+        Arguments:
+            scenario_config_id (str): The identifier of the scenario config used to
+                get the data node rank.
+
+        Returns:
+            The int value representing the rank of the data node config in the scenario config DAG.
+            If the data node config is not found or has no rank, 0xffff is returned. These cases should never happen.
+            If the data node config is not part of the scenario config, 0xfffe is returned as an infinite rank.
+        """
+        if not scenario_config_id:
+            return 0xfffb
+        dn_config = Config.data_nodes.get(self._config_id, None)
+        if not dn_config:
+            self._logger.error(f"Data node config `{self.config_id}` for data node `{self.id}` is not found.")
+            return 0xfffd
+        if not dn_config._ranks:
+            self._logger.error(f"Data node config `{self.config_id}` for data node `{self.id}` has no rank.")
+            return 0xfffe
+        return dn_config._ranks.get(scenario_config_id, 0xfffc)
```

Comment:
Any reason for having changed the returned values? The doc should be also updated.
```suggestion
            The int value representing the rank of the data node config in the scenario config DAG.
            If the scenario config is None or an empty string, 0xfffb is returned.<br/>
            If the data node config is not found, 0xfffd is returned. This case cannot
            happen in a normal situation.<br/>
            If the data node config has no precomputed ranks, 0xfffe is returned. This case
            cannot happen in a normal situation.<br/>
            If the data node config is not part of the scenario config, 0xfffc is returned as an infinite rank.
        """
        if not scenario_config_id:
            return 0xfffb
        dn_config = Config.data_nodes.get(self._config_id, None)
        if not dn_config:
            self._logger.error(f"Data node config `{self.config_id}` for data node `{self.id}` is not found.")
            return 0xfffd
        if not dn_config._ranks:
            self._logger.error(f"Data node config `{self.config_id}` for data node `{self.id}` has no rank.")
            return 0xfffe
        return dn_config._ranks.get(scenario_config_id, 0xfffc)
```

Reply:
made more sense to me